hudi-utilities/src/main/java/org/apache/hudi/utilities/callback/kafka/HoodieWriteCommitKafkaCallback.java

```diff
@@ -0,0 +1,145 @@
+    this.props = config.getProps();
+    this.bootstrapServers = props.getProperty(CALLBACK_KAFKA_BOOTSTRAP_SERVERS);
+    this.topic = props.getProperty(CALLBACK_KAFKA_TOPIC);
+    validateKafkaConfig();
+  }
+
+  @Override
+  public void call(HoodieWriteCommitCallbackMessage callbackMessage) {
+    String callbackMsg = HoodieWriteCommitCallbackUtil.convertToJsonString(callbackMessage);
+    try (KafkaProducer<String, String> producer = createProducer(props)) {
+      ProducerRecord<String, String> record = buildProducerRecord(props, callbackMsg);
+      producer.send(record);
+      LOG.info(String.format("Send callback message %s succeed", callbackMsg));
+    } catch (Exception e) {
+      LOG.error("Send kafka callback msg failed : " + e);
+    }
+  }
+
+  /**
+   * Method helps to create {@link KafkaProducer}. Here we set acks = all and retries = 3 by default to ensure no data
+   * loss.
+   *
+   * @param props Kafka configs
+   * @return A {@link KafkaProducer}
+   */
+  public KafkaProducer<String, String> createProducer(Properties props) {
+    Properties kafkaProducerProps = new Properties();
+    // bootstrap.servers
+    kafkaProducerProps.setProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+    // default "all" to ensure no message loss
+    kafkaProducerProps.setProperty(ProducerConfig.ACKS_CONFIG, props.getProperty(CALLBACK_KAFKA_ACKS));
+    // retries 3 times by default
+    kafkaProducerProps.setProperty(ProducerConfig.RETRIES_CONFIG, props.getProperty(CALLBACK_KAFKA_RETRIES));
+    kafkaProducerProps.setProperty(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
+        "org.apache.kafka.common.serialization.StringSerializer");
+    kafkaProducerProps.setProperty(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
+        "org.apache.kafka.common.serialization.StringSerializer");
+
+    LOG.info("Callback kafka producer init with configs: "
+        + HoodieWriteCommitCallbackUtil.convertToJsonString(kafkaProducerProps));
+    return new KafkaProducer<String, String>(kafkaProducerProps);
+  }
+
+  /**
+   * Method helps to create a {@link ProducerRecord}. To ensure the order of the callback messages, we should guarantee
+   * that the callback message of the same table will goes to the same partition. Therefore, if user does not specify
+   * the partition, we can use the table name as {@link ProducerRecord} key.
+   *
+   * @param props       Kafka configs
+   * @param callbackMsg Callback message
+   * @return Callback {@link ProducerRecord}
+   */
+  private ProducerRecord<String, String> buildProducerRecord(Properties props, String callbackMsg) {
+    String partition = props.getProperty(CALLBACK_KAFKA_PARTITION);
+    if (null != partition) {
+      return new ProducerRecord<String, String>(topic,
+          Integer.valueOf(partition),
+          props.getProperty(TABLE_NAME),
+          callbackMsg);
+    } else {
+      return new ProducerRecord<String, String>(topic,
```

Comment:
Maybe we do not need to break the lines here?

Reply:
done